### PR TITLE
Added sort to document array

### PR DESCRIPTION
**Pull request by sungodavi**
Added a sort function so that the library can be used to model more complicated queries. We could also stub more of the cursor methods to make the package more robust.

### CODE REVIEW

File: lib/sinon-mongo.js [hunk 1]
```diff
@@ -50,9 +50,11 @@ const install = sinon => {
     if (!result) result = [];
     if (result.constructor !== Array) result = [result];  
 
-    return {
+    const cursor = {
```
Comment (DaniJG) [inline lib/sinon-mongo.js]: Thanks for the PR!

I think we should rather setup sort to be a stub that returns itself, as the code under test would typically look like:
```
collection
  .find({name: 'foo'}, {email: 1, name: 1})
  .sort({email:1})
  .toArray();
```

So we can create the stub as
```
const cursor = {
  sort: sinon.stub().returnsThis(),
  toArray: () => Promise.resolve(result)
};
return cursor;
```

That way we can easily test code that uses sort at the same time that we can assert sort was called as expected

**Comment by sungodavi** [comment]
Sorry about that. I meant to type "sinon.fake.resolves(cursor)." Your code looks good.

**Comment by sungodavi** [comment]
I rewrote the method to match your style above. I also added a simple test case for it. Let me know if there's anything else you need to merge

**Comment by DaniJG** [comment]
Thanks, looks good!

**Pull request merged**